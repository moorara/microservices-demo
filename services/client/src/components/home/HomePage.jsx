import React from 'react'
import { Link } from 'react-router-dom'

const HomePage = (props) => (
  <section className="hero is-light is-large is-bold">
    <div className="hero-body">
      <div className="container">
        <h1 className="title">Welcome!</h1>
        <div className="content">
          <p>You are landed on home page a cool application built using cool technologies both in the front and in the end!</p>
          <p>
            <Link to="/sites" className="button is-primary is-medium is-outlined">Sites</Link>
            <span>&nbsp;&nbsp;</span>
            <Link to="/about-us" className="button is-info is-medium is-outlined">About</Link>
          </p>
        </div>
      </div>
    </div>
  </section>
)

export default HomePage
