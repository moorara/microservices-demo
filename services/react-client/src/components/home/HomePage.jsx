import React from 'react'
import { Link } from 'react-router-dom'

const Home = (props) => (
  <section className="hero is-light is-large is-bold">
    <div className="hero-body">
      <div className="container">
        <h1 className="title">Welcome!</h1>
        <div className="columns">
          <div className="column">
            <Link to="/about-us" className="button is-info is-large is-outlined">About</Link>
          </div>
        </div>
      </div>
    </div>
  </section>
)

export default Home
