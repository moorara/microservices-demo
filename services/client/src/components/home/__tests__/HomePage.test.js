import React from 'react'
import { Link } from 'react-router-dom'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import HomePage from '../HomePage'

describe('HomePage', () => {
  test('renders react component', () => {
    const wrapper = shallow(<HomePage />)

    expect(shallowToJson(wrapper)).toMatchSnapshot()
    expect(wrapper.find('.title').exists()).toBeTruthy()
    expect(wrapper.find('.content').exists()).toBeTruthy()
    expect(wrapper.find(Link).find('[to="/sites"]').children().text()).toBe('Sites')
    expect(wrapper.find(Link).find('[to="/about-us"]').children().text()).toBe('About')
  })
})
