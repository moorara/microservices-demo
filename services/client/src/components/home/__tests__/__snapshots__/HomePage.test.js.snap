// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage renders react component 1`] = `
<section
  className="hero is-light is-large is-bold"
>
  <div
    className="hero-body"
  >
    <div
      className="container"
    >
      <h1
        className="title"
      >
        Welcome!
      </h1>
      <div
        className="content"
      >
        <p>
          You are landed on home page a cool application built using cool technologies both in the front and in the end!
        </p>
        <p>
          <Link
            className="button is-primary is-medium is-outlined"
            to="/sites"
          >
            Sites
          </Link>
          <span>
              
          </span>
          <Link
            className="button is-info is-medium is-outlined"
            to="/about-us"
          >
            About
          </Link>
        </p>
      </div>
    </div>
  </div>
</section>
`;
